### ISSUE TITLE
can't query for points before epoch 0

### ISSUE THREAD

**Issue by beckettsean** [influxdata/influxdb#7194]
### Bug report

__System info:__ [Include InfluxDB version, operating system name, and other relevant details]
Ubuntu 14.01, InfluxDB 0.13 from package

__Steps to reproduce:__

Add some data, including data before epoch 0
```
Connected to http://localhost:8086 version 0.13.0
InfluxDB shell version: 0.13.0
> use mydb
Using database mydb
> insert e0 value=1 -100
> insert e0 value=2 100
> insert e0 value=3
```

As expected, epoch 0 is the lower bound if none is provided:
```
> select * from e0
name: e0
--------
time			value
100			2
1471979518382460726	3

> 
```

However, the lower bound cannot be changed to anything less than epoch 0:

```
> select * from e0 where time > '1968-01-01'
ERR: time 1968-01-01T00:00:00Z underflows time literal
> select * from e0 where time > '1968-01-01T00:00:00Z'
ERR: time 1968-01-01T00:00:00Z underflows time literal
> select * from e0 where time > '1970-01-01T00:00:00Z'
name: e0
--------
time	value
100	2

> select * from e0 where time > '1969-01-01T00:00:00Z'
ERR: time 1969-01-01T00:00:00Z underflows time literal
> select * from e0 where time > '1969-12-31T23:59:59Z'
ERR: time 1969-12-31T23:59:59Z underflows time literal
> select * from e0 where time > now() - 4000w
ERR: time 1939-12-26T19:12:10Z underflows time literal
```

__Expected behavior:__ [What you expected to happen]

If negative timestamps are valid for writes, they need to be valid for reads.

**Comment by jwilder** [comment]
This is using `0.9.6.1`. It didn't work either.

```
> insert e0 value=1 -100
> insert e0 value=2 100
> insert e0 value=3
> select * from e0
name: e0
--------
time   				value
1970-01-01T00:00:00.0000001Z   	2
2016-08-24T23:28:02.453424926Z 	3

> select * from e0 where time > '1968-01-01'
> select * from e0 where time > '1968-01-01T00:00:00Z'
> select * from e0 where time > '1970-01-01T00:00:00Z'
name: e0
--------
time   				value
1970-01-01T00:00:00.0000001Z   	2
2016-08-24T23:28:02.453424926Z 	3

> select * from e0 where time > '1969-01-01T00:00:00Z'
> select * from e0 where time > '1969-12-31T23:59:59Z'
> select * from e0 where time > now() - 4000w
```

**Comment by jsternberg** [comment]
So this isn't a regression then? I think we've almost got it implemented, but then it would be a new feature and not a regression.

**Comment by beckettsean** [comment]
Worked in 0.9.1: https://github.com/influxdata/influxdb/issues/2703

Seems to have been part of the 0.9 feature set: https://github.com/influxdata/influxdb/pull/4500.

Not sure why it stopped working with 0.9.6, but clearly we need to decide and document once and for all if we do or do not support pre-`epoch 0` timestamps.

@pauldix @toddboom

**Comment by pauldix** [comment]
I think we should support them unless there's a compelling reason not to. Financial cases bringing in historical market data might want to have data pre 1970.

**Comment by jwilder** [comment]
I agree w/ @pauldix.  Not supporting them would exclude many use cases that use historical data.

**Comment by jsternberg** [comment]
Fixed via #7212.

**Comment by drasko** [comment]
Is this fixed? I've jsut installed 1.2.2 on Debian, and I am getting:

```
2017/04/23 17:26:10 time 1000-01-01T00:00:00Z underflows time literal
```

**Comment by theoDiefenthal** [comment]
@drasko Dates prior to 1.1.1970 do work but not up to 0001. It's more like 1950 is working... 

I don't see a document which points out, which now is the minimum and maximum time?! I guess, the timestamp is stored in a kind of Integer and might be platform dependent. (32/64). Is there a possibility to query my installation about what is the minimum and what is the maximum?
